### PR TITLE
xemu: cosmetic rewrite

### PR DESCRIPTION
**Pull request by AndersonTorres**
Also, add myself as maintainer.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by winterqt** [comment]
But why? You should probably explain what exactly about this is better than the previous version (because it seems like nothing is).

**Comment by AndersonTorres** [comment]
__Reorder the inputs and using `install` instead of `mkdir + cp`__ count?

**Comment by AndersonTorres** [comment]
Why to split the maintainers here? It will not affect possible bisections, after all.

File: pkgs/applications/emulators/xemu/default.nix [hunk 1]
```diff
@@ -112,12 +115,12 @@ stdenv.mkDerivation rec {
...
-    do
-      mkdir -p $out/share/icons/hicolor/$RES/apps/
-      cp ../ui/icons/xemu_$RES.png $out/share/icons/hicolor/$RES/apps/xemu.png
-    done
+  ''
+  + (lib.concatMapStringsSep "\n"
+    (res:
+      "install -Dm644 -T ../ui/icons/xemu_${res}.png $out/share/icons/hicolor/${res}/apps/xemu.png")
+    [ "16x16" "24x24" "32x32" "48x48" "128x128" "256x256" "512x512" ])
+  + ''
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/applications/emulators/xemu/default.nix:123]: I am a bit averse of using Bash, it is unpredictable sometimes. I prefer to treat it as a lower level language, and use higher level constructions from Nix itself to express intent.

**Comment by AndersonTorres** [comment]
@ofborg build xemu

**Pull request merged**